scripts/validate-links.sh

```
@@ -0,0 +1,169 @@
+
+# Function to check if a file exists
+check_internal_link() {
+    local file_path="$1"
+    local source_file="$2"
+    local link="$3"
+    
+    # Remove anchor from path
+    local path_without_anchor="${file_path%%#*}"
+    
+    # Resolve relative path from source file directory
+    local source_dir=$(dirname "$source_file")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)